lib/muchos/config.py

```diff
@@ -130,17 +134,30 @@ def force_format(self):
                 return 'no'
         return retval
 
-    def worker_data_dirs(self):
+    def data_dirs_common(self, nodeType):
+        data_dirs = []
+
         if self.get_cluster_type() == 'ec2':
-            return self.node_type_map()['worker']['mounts']
+            data_dirs = self.node_type_map()[nodeType]['mounts']
         elif self.get_cluster_type() == 'existing':
-            return self.get('existing', 'data_dirs').split(",")
+            data_dirs = self.get('existing', 'data_dirs').split(",")
+        elif self.get_cluster_type() == 'azure':
+            num_disks = int(self.get("azure","numdisks"))
+            range_var = num_disks + 1
+            for diskNum in range(1, range_var):
+                data_dirs.append(self.get("azure","mount_root") + str(diskNum))
+
+        # we need a minimum of 3 "data directories" in order for the HA changes to work correctly. If the number of disks specified is < 3, stub the balance entries.
+        for tmpId in range (len(data_dirs) - 1, 2):
```

Comment:
In the process of the Hadoop / ZK HA changes, what we were recommended was to have separate disks for ZK and Hadoop namenode folders. In this process we ended up with 3 separate folder locations.

In the case of deployments where the ephemeral disk is the only disk used, or in case of Azure when there is just 1 data disk attached, we have to "stub" the extra disk locations. This step does that. I believe post this PR, we will make the entire Hadoop / ZK HA configuration switchable wherein we will restore the old behavior if there is a non-HA config selected.

Reply:
>  I believe post this PR, we will make the entire Hadoop / ZK HA configuration switchable wherein we will restore the old behavior if there is a non-HA config selected.

I would be in favor of that.

Reply:
@arvindshmicrosoft if an issue is opened for this then I can resolve this comment to unclutter the PR.  I would be happy to open the issue unless you would like to.

Reply:
Perfect, we opened #271 to track the next PR which will make the Hadoop / ZK HA config switchable. You may resolve this comment.